private/pkg/git/cloner.go

```diff
@@ -354,7 +358,7 @@ func getRefspecsForName(gitName Name) (fetchRef string, fallbackRef string, chec
 	} else if checkout != "" && checkout != "HEAD" {
 		// If a checkout ref is specified, we fetch the ref directly.
 		// We fallback to fetching the HEAD to resolve partial refs.
-		return checkout, "HEAD", ""
+		return checkout, "HEAD", checkout
```

Comment:
This was invalid before, we should never have a fallback ref without a checkout.